tensorflow/mnist.py

```diff
@@ -10,140 +10,136 @@
-            feed_dict={images: images_data,
-                       labels: labels_data})
-    return g_acc[1]
-
-
-config = tf.ConfigProto(
-    intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
-with tf.Session(config=config) as sess:
-    init_g = tf.global_variables_initializer()
-    init_l = tf.local_variables_initializer()
-    sess.run(init_g)
-    sess.run(init_l)
-    for pass_id in range(PASS_NUM):
-        pass_start = time.time()
-        for batch_id, data in enumerate(train_reader()):
+        paddle.dataset.mnist.test(), batch_size=args.batch_size)
+
+    def eval_test():
+        for batch_id, data in enumerate(test_reader()):
             images_data = np.array(
                 map(lambda x: np.transpose(x[0].reshape([1, 28, 28]), axes=[1,2,0]), data)).astype("float32")
             labels_data = np.array(map(lambda x: x[1], data)).astype("int64")
-            start = time.time()
+
             _, loss, acc, g_acc = sess.run(
                 [train_op, avg_cost, accuracy, g_accuracy],
                 feed_dict={images: images_data,
                            labels: labels_data})
-            end = time.time()
-
-            print("pass=%d, batch=%d, loss=%f, error=%f, elapse=%f" %
-                  (pass_id, batch_id, loss, 1 - acc, (end - start) / 1000))
-        pass_end = time.time()
-        test_avg_acc = eval_test()
-        print("pass=%d, training_avg_accuracy=%f, test_avg_acc=%f, elapse=%f" %
-              (pass_id, g_acc[1], test_avg_acc, (pass_end - pass_start) / 1000))
+        return g_acc[1]
+
+    config = tf.ConfigProto(
+        intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
+    with tf.Session(config=config) as sess:
+        init_g = tf.global_variables_initializer()
+        init_l = tf.local_variables_initializer()
+        sess.run(init_g)
+        sess.run(init_l)
+        for pass_id in range(args.pass_num):
+            pass_start = time.time()
+            for batch_id, data in enumerate(train_reader()):
+                images_data = np.array(
+                    map(lambda x: np.transpose(x[0].reshape([1, 28, 28]), axes=[1,2,0]), data)).astype("float32")
+                labels_data = np.array(map(lambda x: x[1], data)).astype(
+                    "int64")
+                start = time.time()
+                _, loss, acc, g_acc = sess.run(
+                    [train_op, avg_cost, accuracy, g_accuracy],
+                    feed_dict={images: images_data,
+                               labels: labels_data})
+                end = time.time()
+
+                print("pass=%d, batch=%d, loss=%f, error=%f, elapse=%f" %
+                      (pass_id, batch_id, loss, 1 - acc, (end - start) / 1000))
+            pass_end = time.time()
+            test_avg_acc = eval_test()
+            print(
+                "pass=%d, training_avg_accuracy=%f, test_avg_acc=%f, elapse=%f"
+                % (pass_id, g_acc[1], test_avg_acc,
```

Comment:
Same as above, I think better to reset the accumulating state in the begin of each epoch.

Reply:
Donw